testFlight1/lib/teensyShared/rocket.cpp

```diff
@@ -0,0 +1,244 @@
+                rocketTimers.burnout_time = chVTGetSystemTime();
+                pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_BURNOUT_DETECT;
+            }
+ 
+            break;
+ 
+            case STATE_BURNOUT_DETECT:
+ 
+                //If the 0 acceleration was too brief, go back to BOOST
+                if (pointer_struct->sensorDataPointer->state_data.state_az > coast_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_BOOST;
+                    break;
+                }
+ 
+                // measure the length of the coast time (for hysteresis)
+                rocketTimers.coast_timer =
+                    chVTGetSystemTime() - rocketTimers.burnout_time;
+ 
+                // If the low acceleration lasts long enough, coast is detected
+                if (rocketTimers.coast_timer > coast_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_COAST;
+                }
+ 
+            break;
+ 
+            case STATE_COAST:
+                //if velocity <= 0
+                if (pointer_struct->sensorDataPointer->state_data.state_vz <= apogee_thresh) {
+                    rocketTimers.apogee_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE_DETECT;
+                }
+ 
+            break;
+ 
+            case STATE_APOGEE_DETECT:
+                //If the negative velocity was too brief, go back to coast
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > apogee_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_COAST;
+                    break;
+                }
+ 
+                // measure the length of the apogee time
+                rocketTimers.apogee_timer =
+                    chVTGetSystemTime() - rocketTimers.apogee_time;
+ 
+                // If the < 0 velocity lasts long enough, apogee is detected
+                if (rocketTimers.coast_timer > apogee_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE;
+                }
+            break;
+ 
+            case STATE_APOGEE: //*assumes velocity is <= 0
+                //If rocket has velocity in main range
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < main_thresh_max
+                        && pointer_struct->sensorDataPointer->state_data.state_vz > main_thresh_min) {
+                    rocketTimers.main_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_MAIN_DETECT;
+                }
+                //If rocket has volocity in drogue range
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < drogue_thresh) {
```

Comment:
I believe an `else if` is necessary here. right?

Reply:
agreed